Jellyfin.Server.Implementations/Devices/DeviceManager.cs

```diff
@@ -108,57 +125,42 @@ public ClientCapabilities GetCapabilities(string deviceId)
         }
 
         /// <inheritdoc />
-        public async Task<DeviceInfo?> GetDevice(string id)
+        public DeviceInfo? GetDevice(string id)
         {
-            var dbContext = await _dbProvider.CreateDbContextAsync().ConfigureAwait(false);
-            await using (dbContext.ConfigureAwait(false))
-            {
-                var device = await dbContext.Devices
-                    .Where(d => d.DeviceId == id)
-                    .OrderByDescending(d => d.DateLastActivity)
-                    .Include(d => d.User)
-                    .SelectMany(d => dbContext.DeviceOptions.Where(o => o.DeviceId == d.DeviceId).DefaultIfEmpty(), (d, o) => new { Device = d, Options = o })
-                    .FirstOrDefaultAsync()
-                    .ConfigureAwait(false);
-
-                var deviceInfo = device is null ? null : ToDeviceInfo(device.Device, device.Options);
+            var device = _devices.Values.OrderByDescending(d => d.DateLastActivity).FirstOrDefault(d => d.DeviceId == id);
+            _deviceOptions.TryGetValue(id, out var deviceOption);
 
-                return deviceInfo;
-            }
+            var deviceInfo = device is null ? null : ToDeviceInfo(device, deviceOption);
+            return deviceInfo;
         }
 
         /// <inheritdoc />
-        public async Task<QueryResult<Device>> GetDevices(DeviceQuery query)
+        public QueryResult<Device> GetDevices(DeviceQuery query)
         {
-            var dbContext = await _dbProvider.CreateDbContextAsync().ConfigureAwait(false);
-            await using (dbContext.ConfigureAwait(false))
+            IEnumerable<Device> devices = _devices.Values.OrderBy(d => d.Id)
+                .Where(device => !query.UserId.HasValue || device.UserId.Equals(query.UserId.Value))
+                .Where(device => query.DeviceId == null || device.DeviceId == query.DeviceId)
+                .Where(device => query.AccessToken == null || device.AccessToken == query.AccessToken)
+                .ToList();
+            var count = devices.Count();
```

Comment:
```suggestion
            var count = devices.Count;
```

Reply:
We cannot do this as devices is a generic delegate, maybe `var count = ((List<Device>)devices).Count;` will do the trick?

Reply:
Ah right, I just remembered the `.ToList()`. Updating the `.Count()` is unneeded then